### ISSUE TITLE
False Positive UnnecessaryApply

### ISSUE THREAD

**Issue by vanniktech** [detekt/detekt#1539]
Following code gets flagged:

```kotlin
SpannableStringBuilder(text).apply { setSpan(ForegroundColorSpan(textColor), 0, length, Spanned.SPAN_INCLUSIVE_INCLUSIVE) }
```

`setSpan` has return type `void` and hence it's not unnecessary

Using RC14.

**Comment by 3flex** [comment]
I'm looking at some of the previous issues that were waiting for type resolution.

I'm confused by the original issue (I'm sure it's just me!): in this case, is the `SpannableStringBuilder` being assigned to a property?

If it's `val ssb = SpannableStringBuilder(text).apply { setSpan(...) }` I can see how that should not be flagged, but if the line of code is `SpannableStringBuilder(text).apply { setSpan(...) }` I'm not sure it makes sense since it has to be assigned to something to be used?

I'd have to look into why, but there's already a test to make sure `val content = Intent().apply { putExtra("", 1) }` doesn't report, so wouldn't that cover the first case above?

Again, I'm sure I'm missing something, so sorry if that's the case :(

**Comment by 3flex** [comment]
Perhaps the rule should ignore usage of `apply` when it's used on a newly instantiated instance of a class?

**Comment by vanniktech** [comment]
Not really. Sometimes apply is unnecessary on a newly instantiated instance.

**Comment by 3flex** [comment]
I agree, but I'm not sure how else we could avoid this false positive. Is there another way it could be done?

**Comment by marschwar** [comment]
IMO this is solved by now and can be closed.
